src/app/shared/expiration/date-picker/expiration-date.component.ts

```diff
@@ -78,15 +78,12 @@ export class ExpirationDateComponent implements OnInit, OnDestroy {
   }
 
   restoreDefaultHandler(): void {
+    this.defaultExpirationDate();
     this.add.emit(this.defaultValidityPeriod);
```

Comment:
This is a bit odd to me. `expirationDate` is being set but we're emitting `defaultValidityPeriod`. But maybe I'm missing some pattern/convention here. Does it make sense to you?

Reply:
Yeah, defaultExpirationDate was setting expirationDate to be equal default, so it worked, but logically it wasn't correct. Changed it by adding method:
```
  private emitExpirationTimeShift() {
    this.add.emit(this.expirationTimeShift);
  }
```